sound/soc/sof/intel/hda-dai.c

```diff
@@ -483,9 +462,20 @@ static int ssp_dai_hw_free(struct snd_pcm_substream *substream,
 	return ssp_dai_setup_or_free(substream, dai, false);
 }
 
+/* handle SUSPEND trigger to free the DAI widget during suspend */
+static int ssp_dai_trigger(struct snd_pcm_substream *substream,
+			   int cmd, struct snd_soc_dai *dai)
+{
+	if (cmd == SNDRV_PCM_TRIGGER_SUSPEND || cmd == SNDRV_PCM_TRIGGER_STOP)
+		return ssp_dai_setup_or_free(substream, dai, false);
```

Comment:
erm, what?

We can do this for suspend but not for stop.
This will completely break the work for the SSP early/late clocks.

i stop the review here, this is a show-stopper for me.